### PR TITLE
Add pacbio sequencing run endpoint

### LINKED ISSUE #4161: New endpoints needed for fetching PacBio sequencing data
As a LIMS developer,
I want new endpoints in cg for fetching PacBio data,
So that we can set up post-sequencing QC checks in LIMS.

# Work impact

Answer the following questions:
- Is there currently a workaround for this issue? If so, what is it?
    - Nope.
- How much time would be saved by implementing this feature on a weekly basis?
    - Not a time-saving issue.
- How many users are affected by this issue?
    - All LIMS developers and by proxy prod bioinfo
- Are customers affected by this issue?
    - Nope.

# Acceptance Criteria

- [ ] GET `/pacbio_sequencing_run/{run_name}` returns `list[PacbioSequencingRun]`
- [x] GET `/pacbio_sample_sequencing_metrics?sample=<sample_id>&smrt_cell=<cell_id>` returns `list[PacbioSampleSequencingMetrics]`

# Notes

- Additional information.
- Dependencies.
- Related user stories.

### PR DESCRIPTION
**Pull request by islean**
## Description

This PR adds the final endpoint and closes #4161.

### Added

- Endpoint for fetching Pacbio sequencing runs based on run name

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/store_helpers.py [hunk 1]
```diff
@@ -1109,3 +1111,62 @@ def add_illumina_flow_cell_and_samples_with_sequencing_metrics(
...
+    def add_pacbio_sequencing_run(
+        store: Store,
+        id: int,
+        device_id: int,
+        well: str = "A01",
+        plate: int = 1,
+        run_name: str = "run_name",
+        movie_name: str = "movie_name",
+    ) -> PacbioSequencingRun:
+        run = PacbioSequencingRun(
```
Comment (islean) [inline tests/store_helpers.py:1132]: No idea how to write these and what data to make settable. Also no idea what data is reasonable to set as defaults.

File: cg/server/endpoints/sequencing_run/pacbio_sequencing_run.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+from flask import Blueprint, jsonify
+
+from cg.server.endpoints.error_handler import handle_missing_entries
+from cg.server.endpoints.sequencing_run.dtos import PacbioSequencingRunsResponse
+from cg.server.endpoints.utils import before_request
+from cg.server.ext import pacbio_sequencing_runs_service
+
+PACBIO_SEQUENCING_RUN_BLUEPRINT = Blueprint("pacbio_sequencing_run", __name__, url_prefix="/api/v1")
```
Comment (ChrOertlin) [inline cg/server/endpoints/sequencing_run/pacbio_sequencing_run.py:8]:
```suggestion
PACBIO_SEQUENCING_RUN_BLUEPRINT = Blueprint("pacbio_sequencing_run", __name__, url_prefix="/api/v1/pacbio_sequencing_run")
```

File: cg/server/endpoints/sequencing_run/pacbio_sequencing_run.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+PACBIO_SEQUENCING_RUN_BLUEPRINT.before_request(before_request)
+
+
+@PACBIO_SEQUENCING_RUN_BLUEPRINT.route("/pacbio_sequencing_run/<run_name>", methods=["GET"])
```
Comment (ChrOertlin) [inline cg/server/endpoints/sequencing_run/pacbio_sequencing_run.py:12]:
```suggestion
@PACBIO_SEQUENCING_RUN_BLUEPRINT.route("/<run_name>", methods=["GET"])
```

File: tests/services/run_devices/pacbio/test_pacbio_sequencing_runs_service.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from cg.server.endpoints.sequencing_run.dtos import PacbioSequencingRunsResponse
+from cg.services.run_devices.pacbio.sequencing_runs_service import PacbioSequencingRunsService
+from cg.store.models import PacbioSequencingRun
+
```
Comment (ChrOertlin) [inline tests/services/run_devices/pacbio/test_pacbio_sequencing_runs_service.py:4]: add test when none present should raise the error

**Review by ChrOertlin — APPROVED** [review]
looks good

File: cg/services/run_devices/pacbio/sequencing_runs_service.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+from cg.server.endpoints.sequencing_run.dtos import (
+    PacbioSequencingRunDTO,
+    PacbioSequencingRunsResponse,
+)
+from cg.store.models import PacbioSequencingRun
+from cg.store.store import Store
+
+
+class PacbioSequencingRunsService:
```
Comment (diitaz93) [inline cg/services/run_devices/pacbio/sequencing_runs_service.py:9]: I insist on this being in the web services directory, but I guess this is related to a bigger discussion about the whole structure of the `services` directory. At this point it is a matter of taste and we don't have clear guidelines, but I want to note that right now we have services that do essentially the same work for different tables (fetching from the database and converting into responses) spread over different directories in an unstructured way.

File: cg/server/endpoints/sequencing_run/pacbio_sequencing_run.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+PACBIO_SEQUENCING_RUN_BLUEPRINT = Blueprint(
+    "pacbio_sequencing_run", __name__, url_prefix="/api/v1/pacbio_sequencing_run"
+)
+PACBIO_SEQUENCING_RUN_BLUEPRINT.before_request(before_request)
+
+
+@PACBIO_SEQUENCING_RUN_BLUEPRINT.route("/<run_name>", methods=["GET"])
+@handle_missing_entries
+def get_sequencing_metrics(run_name: str):
```
Comment (diitaz93) [inline cg/server/endpoints/sequencing_run/pacbio_sequencing_run.py:16]:
```suggestion
def get_sequencing_runs(run_name: str):
```

File: tests/fixture_plugins/pacbio_fixtures/name_fixtures.py [hunk 7]
```diff
@@ -78,3 +78,13 @@ def pacbio_barcoded_sample_internal_id() -> str:
...
+
+
+@pytest.fixture
+def pacbio_run_name_to_fetch() -> str:
+    return "run_to_fetch"
+
+
+@pytest.fixture
+def pacbio_run_name_to_not_fetch() -> str:
+    return "run_to_not_fetch"
```
Comment (diitaz93) [inline tests/fixture_plugins/pacbio_fixtures/name_fixtures.py:90]: 🤓 
```suggestion
    return "run_not_to_fetch"
```

File: tests/store_helpers.py (hunk 1 above)
Comment (diitaz93) [reply tests/store_helpers.py:1132]: see https://github.com/Clinical-Genomics/cg/blob/fa45dce79f8d4d7d0535f0a59b1b9ab874434788/tests/fixture_plugins/pacbio_fixtures/dto_fixtures.py#L20 as an inspiration. JSON idea sounds good to reduce the number of parameters

**Review by diitaz93 — APPROVED** [review]
Looks good 💯

File: tests/services/run_devices/pacbio/test_pacbio_sequencing_runs_service.py (hunk 4 above)
Comment (islean) [reply tests/services/run_devices/pacbio/test_pacbio_sequencing_runs_service.py:7]: Done

File: tests/store_helpers.py (hunk 1 above)
Comment (islean) [reply tests/store_helpers.py:1132]: I can just remove the parameters that I do not set here. The only things I need control over are the id, the device_id, and the store

File: cg/services/run_devices/pacbio/sequencing_runs_service.py (hunk 5 above)
Comment (islean) [reply cg/services/run_devices/pacbio/sequencing_runs_service.py:9]: I agree but I think it is a separate discussion, and I do not think that `web_services` is beneficial, since it seems to suggest that that is their intended place of usage. I think this also ties into me thinking that the store is missing a service layer as a whole. I would probably call these store services or something. But let us discuss it in during coding guidelines or something?

File: tests/store_helpers.py (hunk 1 above)
Comment (diitaz93) [reply tests/store_helpers.py:1132]: sounds good

**Comment by islean** [comment]
Deployed to the VMs
